src/Cli/Microsoft.DotNet.Cli.Utils/Extensions/ProcessExtensions.cs

```diff
@@ -36,27 +33,19 @@ public static Process GetParentProcess(this Process process)
         /// </summary>
         /// <param name="process">The process component.</param>
         /// <returns>The process ID of the parent process, or -1 if the parent process could not be found.</returns>
-        public static int GetParentProcessId(this Process process)
+        public unsafe static int GetParentProcessId(this Process process)
         {
-            CimSession cimSession = CimSession.Create(null);
-            IEnumerable<CimInstance> results = cimSession.QueryInstances(@"root\cimv2", "WQL",
-                $"SELECT ParentProcessId FROM Win32_Process WHERE ProcessId='{process.Id}'");
+            using SafeProcessHandle handle = new(process.Handle, ownsHandle: false);
```

Comment:
@jkotas Is this call correct or should I duplicate the GetProcessHandle logic?

Reply:
```suggestion
            SafeProcessHandle handle = process.SafeHandle;
```

Reply:
>  should I duplicate the GetProcessHandle logic?

I do not think you need to duplicate `GetProcessHandle` logic. SafeHandle property should have all permissions that it can have given how you are using it.

Reply:
fixed